Frontend/src/Components/Faults/FaultsView.js

```diff
@@ -285,14 +285,217 @@ export default function Faults(props) {
     );
   };
 
+  const getHeartbeatString = () => {
+    let heartbeatStrings = [];
+
+    // Add any failing heartbeats
+    // The first element pushed should be capitalized, and any
+    // additional failing heartbeats should not be capitalized
+    if (_checkBooleanData("mainIO_heartbeat")) {
+      heartbeatStrings.push(heartbeatStrings.length === 0 ? "Main IO" : "main IO");
+    }
+    if (_checkBooleanData("mppt_can_heartbeat")) {
+      heartbeatStrings.push("MPPT CAN");
+    }
+    if (_checkBooleanData("mcc_can_heartbeat")) {
+      heartbeatStrings.push("MCC CAN");
+    }
+    if (_checkBooleanData("bms_can_heartbeat")) {
+      heartbeatStrings.push("BMS CAN");
+    }
+    
+    
+    // Create and return a formatted string containing a list of hearbeat failures
+    return _formFaultString(
+      heartbeatStrings,
+      " Heartbeat was not detected",
+      " Heartbeats were not detected"
+    );
+  };
+
+  const getMCUString = () => {
+    let MCUStrings = [];
+
+    if (_checkBooleanData("mcu_check")) {
+      MCUStrings.push("MCU check");
+    }
+    if (_checkBooleanData("mcu_stat_fdbk")) {
+      MCUStrings.push("MCU status feedback");
+    }
+    if (_checkBooleanData("mcu_hv_en")) {
+      MCUStrings.push("MCU status");
+    }
+    
+    // Create and return a formatted string containing a list of MCU errors
+    return _formFaultString(
+      MCUStrings,
+      " error was found",
+      " errors were found"
+    );
+  };
+
+  const getMainDriverPowerString = () => {
+    let mainDriverPowerStrings = [];
+
+    // Add any erroneous main or driver power values to the tooltip
+    // The first element pushed should be capitalized, and any additional
+    // parts of the string should not be
+    if (_checkBooleanData("main_power_warning")) {
+      mainDriverPowerStrings.push(mainDriverPowerStrings.length === 0 ? "Main power warning" : "main power warning");
+    }
```

Comment:
`mainDriverPowerStrings.length` can't be greater than 0, so we can default to the capitalized string